### PR TITLE
fix bugs on non existent vs no reserves, fix search scanning, small UI changes

### PR DESCRIPTION
**Pull request by ianlapham**
- for usePair: return undefined if loading, null if no pair created, and pair otherwise
- fix QR code scanning on dark mode

### CODE REVIEW

File: src/components/CreatePool/index.tsx [hunk 1]
```diff
@@ -35,16 +35,14 @@ function CreatePool({ history }) {
   const [step, setStep] = useState<number>(1)
 
   const pair = usePair(token0, token1)
-  const pairExists = // used to detect new exchange
-    pair && JSBI.notEqual(pair.reserve0.raw, JSBI.BigInt(0)) && JSBI.notEqual(pair.reserve1.raw, JSBI.BigInt(0))
 
   useEffect(() => {
-    if (token0Address && token1Address && pair && !pairExists) {
+    if (token0Address && token1Address && !pair) {
```
Comment (NoahZinsmeister) [inline src/components/CreatePool/index.tsx:40]: hm, what's the logic here? a bit hard to parse what the steps represent, a few comments here and there might helpful

File: src/components/SearchModal/index.tsx [hunk 2]
```diff
@@ -184,6 +184,19 @@ function SearchModal({
...
   const [temporaryToken, setTemporaryToken] = useState<Token | null>()
+
+  // filters for ordering
+  const [activeFilter, setActiveFilter] = useState(FILTERS.BALANCES)
+
+  // toggle specific token import view
+  const [showTokenImport, setShowTokenImport] = useState(false)
+
+  // for sorting
+  const escapeStringRegexp = string => string
```
Comment (NoahZinsmeister) [inline src/components/SearchModal/index.tsx:195]: this should be: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#Escaping

Comment (NoahZinsmeister) [reply src/components/SearchModal/index.tsx:195]: let's just make that a util function, make sure we're not doing this anywhere else, and remove the `escape-string-regexp` dependency

File: src/components/CreatePool/index.tsx (hunk 1 above)
Comment (ianlapham) [reply src/components/CreatePool/index.tsx:40]: Yep replaced this with more clear steps for the views 

- one thing im considering is just linking straight to the regular add liquidity page. The only reason we keep the AddLiquidity component within this flow is so on the nav we can say "Create Pool" instead of 'Add Liquidity" on the last step (based on the route)

File: src/components/SearchModal/index.tsx (hunk 2 above)
Comment (ianlapham) [reply src/components/SearchModal/index.tsx:195]: Yep adding this to utils

File: src/components/CreatePool/index.tsx (hunk 1 above)
Comment (ianlapham) [reply src/components/CreatePool/index.tsx:47]: Another thing is - at some point we discussed that this flow is probably best to let them pick tokens and then add - but should there ever be an option to just create the pool without adding? currently you cant do that in the UI

Comment (haydenadams) [reply src/components/CreatePool/index.tsx:47]: > but should there ever be an option to just create the pool without adding? currently you cant do that in the UI

prob? kinda nice to cover the gas cost for someone else I guess if its like $5

Comment (ianlapham) [reply src/components/CreatePool/index.tsx:47]: for new pools we could maybe popup a smaller option that says "create without liquidity" or something (on the add page)

Comment (ianlapham) [reply src/components/CreatePool/index.tsx:47]: idk if the added confusion is worth helping the few people who want to create and not add though
